spec/unit/policyfile_evaluation_spec.rb

```diff
@@ -323,6 +323,63 @@
 
     end
 
+    describe "defining attributes" do
+
+      let(:policyfile_rb) do
+        <<-EOH
+          name "policy-with-attrs"
+          run_list "foo"
+
+          # basic attribute setting:
+          default["foo"] = "bar"
+
+          # auto-vivify
+          default["abc"]["def"]["ghi"] = "xyz"
+
+          # literal data structures
+          default["baz"] = {
+            "more_nested_stuff" => "yup"
+          }
+
+          # Array literals work and we merge rather than overwrite:
+          default["baz"]["an_array"] = ["a", "b", "c"]
+
+          # all the same stuff works with overrides:
+
+          override["foo"] = "bar"
+
+          override["abc"]["def"]["ghi"] = "xyz"
+
+          override["baz"] = {
+            "more_nested_stuff" => "yup"
+          }
+
+          override["baz"]["an_array"] = ["a", "b", "c"]
+        EOH
+      end
+
+      let(:expected_merged_attrs) do
+        {
+          "foo" => "bar",
+          "abc" => { "def" => { "ghi" => "xyz" } },
+          "baz" => {
+            "more_nested_stuff" => "yup",
+            "an_array" => ["a", "b", "c"]
+          }
+        }
+      end
+
+      it "defines default attributes" do
+        expect(policyfile.errors).to eq([])
+        expect(policyfile.default_attributes).to eq(expected_merged_attrs)
```

Comment:
Probably the confusion is because I chose my wording poorly and/or because I was lazy and used the same inputs for default and override. Anyway, the "merge" part is not that default and override are merged, just that all of the attributes given on different lines of code in the Policyfile.rb get combined in the normal way that Chef Client would combine them in an attributes file. We store default and override separately for the same reason we do so for roles right now--you typically will set different things at the default/override levels. I'll clean up the example to make this more obvious.